module/preference/PreferenceDialog.kt

```diff
@@ -0,0 +1,386 @@
+import com.vaticle.typedb.studio.framework.common.theme.Theme
+import com.vaticle.typedb.studio.framework.material.Dialog
+import com.vaticle.typedb.studio.framework.material.Form
+import com.vaticle.typedb.studio.framework.material.Form.CaptionSpacer
+import com.vaticle.typedb.studio.framework.material.Form.State
+import com.vaticle.typedb.studio.framework.material.Form.RowSpacer
+import com.vaticle.typedb.studio.framework.material.Form.ColumnSpacer
+import com.vaticle.typedb.studio.framework.material.Form.Field
+import com.vaticle.typedb.studio.framework.material.Form.Checkbox
+import com.vaticle.typedb.studio.framework.material.Form.TextInput
+import com.vaticle.typedb.studio.framework.material.Form.TextButton
+import com.vaticle.typedb.studio.framework.material.Form.Text
+import com.vaticle.typedb.studio.framework.material.Frame
+import com.vaticle.typedb.studio.framework.material.Navigator
+import com.vaticle.typedb.studio.framework.material.Navigator.rememberNavigatorState
+import com.vaticle.typedb.studio.framework.material.Separator
+import com.vaticle.typedb.studio.state.StudioState
+import com.vaticle.typedb.studio.state.common.util.Label.APPLY
+import com.vaticle.typedb.studio.state.common.util.Label.CANCEL
+import com.vaticle.typedb.studio.state.common.util.Label.ENABLE_EDITOR_AUTOSAVE
+import com.vaticle.typedb.studio.state.common.util.Label.ENABLE_GRAPH_OUTPUT
+import com.vaticle.typedb.studio.state.common.util.Label.GRAPH_VISUALISER
+import com.vaticle.typedb.studio.state.common.util.Label.MANAGE_PREFERENCES
+import com.vaticle.typedb.studio.state.common.util.Label.OK
+import com.vaticle.typedb.studio.state.common.util.Label.PROJECT_IGNORED_PATHS
+import com.vaticle.typedb.studio.state.common.util.Label.PROJECT_MANAGER
+import com.vaticle.typedb.studio.state.common.util.Label.QUERY_RUNNER
+import com.vaticle.typedb.studio.state.common.util.Label.SET_QUERY_LIMIT
+import com.vaticle.typedb.studio.state.common.util.Label.TEXT_EDITOR
+import com.vaticle.typedb.studio.state.common.util.Sentence.PREFERENCES_GRAPH_OUTPUT_CAPTION
+import com.vaticle.typedb.studio.state.common.util.Sentence.PREFERENCES_MATCH_QUERY_LIMIT_CAPTION
+import com.vaticle.typedb.studio.state.page.Navigable
+import java.util.Optional
+
+object PreferenceDialog {
+    private val WIDTH = 800.dp
+    private val HEIGHT = 600.dp
+    private val NAVIGATOR_INIT_SIZE = 200.dp
+    private val NAVIGATOR_MIN_SIZE = 150.dp
+    private val PREFERENCE_GROUP_INIT_SIZE = 600.dp
+    private val PREFERENCE_GROUP_MIN_SIZE = 500.dp
+
+    private val preferenceMgr = StudioState.preference
+    private var focusedPreferenceGroup by mutableStateOf(PreferenceGroup(""))
+
+    sealed interface PreferenceField {
+        val label: String
+        val caption: Optional<String>
+        fun isValid(): Boolean
+        @Composable fun Display()
+
+        @Composable
+        fun FieldLayout(fieldContent: @Composable () -> Unit) {
```

Comment:
Repetition of `Field` (`PreferenceField.FieldLayout`) - let's just call this `Layout`?